platform: backup to another region (BTAR) docs (#1798)
aiven/devportal: docs/platform/concepts/backup-to-another-region.rst

If you want to have your service restored from a secondary backup, for example, in case of an outage of the primary region, you create a fork of the service using a region where your secondary backup is located.

.. mermaid::

    flowchart LR
        subgraph Primary_region
            direction TB
            subgraph Service_X
                end
            subgraph Primary_backups
                direction LR
                PB1
                PB2
                PB3
                PBn
                end
            end
        subgraph Secondary_region
            direction TB
            subgraph Forked_service_X
                end
            subgraph Secondary_backups
                direction LR
                SB1
                SB2
                SB3
                SBn
                end
            end
        Service_X -- Default \n backups --> Primary_backups
        Primary_backups -- Cross-region backups \n if BTAR  enabled --> Secondary_backups
        Secondary_backups -- Secondary backups \n to restore service X --> Forked_service_X
        Service_X -- Forking service X \n if primary region down --> Forked_service_X

Limitations
-----------

> * When selecting a cloud region for your additional backup, you need to use the same cloud provider that your service uses.

Reviewer: a few words about:  "this is a temporary limitation, about to be lifted away soon" would be good

> * When you want to :ref:`restore your service from an additional backup <fork-and-restore>` and you use a point in time to specify the scope of data to be backed up, you need to set up the time to no later than the time of taking the latest backup.

Reviewer: ```suggestion
* When you want to :ref:`restore your service from an additional backup <fork-and-restore>` and you use a point in time to specify the scope of data to be restored, you need to set up the time to no later than the time of latest backup.
```

this is actually a limitation, that only upto the latest backup can be restored (from secondary location) and nothing after that. This limitation will also be lifted in future

About backup to another region for Aiven services
=================================================

.. important::

    Backup to another region (BTAR) is available on ``:doc:Pro Platform </docs/platform/concepts/pro-platform>``.

Discover the backup to another region (BTAR) feature, check why use it, and learn how it works.

About BTAR
----------

> The backup to another region (BTAR) feature allows existing backup files to be copied from the service's primary backup region to one or more regions different from the primary one. As a disaster recovery feature, BTAR is particularly useful when the service-hosting region is down since it allows forking the service from an additional copy of the backup residing outside the service-hosting region.

Reviewer 2: ```suggestion
The backup to another region (BTAR) feature allows existing backup files to be copied from the service's primary backup region to an additional region. As a disaster recovery feature, BTAR is particularly useful when the service-hosting region is down since it allows forking the service from an additional copy of the backup residing outside the service-hosting region.
```

We may or may not allow multiple regions in the future but we don't at the moment.

BTAR is currently supported for the following services:

* Aiven for PostgreSQL®
* Aiven for MySQL®

Why use BTAR
------------

> BTAR provides an additional level of security for your data. It is a cost-efficient low-complexity disaster recovery feature that allows you to spread your data across different regions and recovery it in case of data loss or corruption in the primary region. During an outage or a downtime on the primary region, your data is protected and safe in a secondary backup location.

Reviewer 2: ```suggestion
BTAR provides an additional level of resilience for your data. It is a cost-efficient, low-complexity disaster recovery feature that allows you to replicate your data in a different region and recover there in case of outage, downtime, data loss or corruption in the primary region.
```

How BTAR works
--------------

> By enabling the backup to another region (BTAR) feature, you create an additional backup on top of the default one located in a primary region where your service is hosted. Contrary to the primary backup, the secondary backup is located outside the primary region.

Reviewer 2: ```suggestion
By enabling the backup to another region (BTAR) feature, you create an additional backup in addition to the default one. This secondary backup is always located in a secondary region of your choice.
```

> Secondary backups are taken from primary backups, not from the service itself. For this reason, your secondary backup is available only after a primary backup is there. There might be a data replication lag between the primary region and the secondary region.

Reviewer 2: ```suggestion
Secondary backups are generated from primary backups, not from the service itself. For this reason, your secondary backup becomes available only after the primary backup is taken. There might be a data replication lag between the primary region and the secondary region.
```

> If you want to have your service restored from a secondary backup, for example, in case of an outage of the primary region, you create a fork of the service using a region where your secondary backup is located.

Reviewer 2: ```suggestion
Restoring from a secondary backup, for example in case of an outage of the primary region, can be done by creating a fork of the service in the region where the secondary backup is located.
```

.. mermaid::

    flowchart LR
        subgraph Primary_region
            direction TB
            subgraph Service_X
                end
            subgraph Primary_backups
                direction LR
                PB1
                PB2
                PB3
                PBn
                end
            end
        subgraph Secondary_region
            direction TB
            subgraph Forked_service_X
                end
            subgraph Secondary_backups
                direction LR
                SB1
                SB2
                SB3
                SBn
                end
            end
        Service_X -- Default \n backups --> Primary_backups
        Primary_backups -- Cross-region backups \n if BTAR  enabled --> Secondary_backups
        Secondary_backups -- Secondary backups \n to restore service X --> Forked_service_X
        Service_X -- Forking service X \n if primary region down --> Forked_service_X

Limitations
-----------

> * When selecting a cloud region for your additional backup, you need to use the same cloud provider that your service uses (support for creating cross-cloud-provider backups coming soon!).

Reviewer 2: ```suggestion
* When selecting a cloud region for your additional backup, you need to use the same cloud provider that your service uses.
```

* When you want to :ref:`restore your service from an additional backup <fork-and-restore>` and you use a point in time to specify the scope of data to be restored, you need to set up the time to no later than the time of the latest backup.

> * To restore a service from an additional (not default) backup, you need to create a fork of the original service in the region where the additional backup resides.

Reviewer 2: ```suggestion
* Secondary backups can only be restored in the region where they were stored.
```

> * Secondary backup is available only after a primary backup is there and there might be a data replication lag between the primary region and the secondary region.

Reviewer 2: ```suggestion
* Secondary backup is generated only after a primary backup is complete and there might be a data replication lag between the primary region and the secondary region.
```
